src/EFCore.PG/Query/Internal/NpgsqlEvaluatableExpressionFilter.cs

```diff
@@ -10,23 +37,46 @@
 // this (we're currently replacing an internal service
 namespace Npgsql.EntityFrameworkCore.PostgreSQL.Query.Internal
 {
+    /// <summary>
+    /// Represents an Npgsql-specific filter to identify methods that are evaluatable.
+    /// </summary>
     public class NpgsqlEvaluatableExpressionFilter : RelationalEvaluatableExpressionFilter
     {
-        static readonly MethodInfo _tsQueryParse = typeof(NpgsqlTsQuery).GetMethod(
-            nameof(NpgsqlTsQuery.Parse),
-            BindingFlags.Public | BindingFlags.Static);
-
-        static readonly MethodInfo _tsVectorParse = typeof(NpgsqlTsVector).GetMethod(
-            nameof(NpgsqlTsVector.Parse),
-            BindingFlags.Public | BindingFlags.Static);
-
-        public NpgsqlEvaluatableExpressionFilter([NotNull] IModel model) : base(model) { }
-
-        public override bool IsEvaluatableMethodCall(MethodCallExpression methodCallExpression) =>
-            methodCallExpression.Method != _tsQueryParse
-            && methodCallExpression.Method != _tsVectorParse
-            && methodCallExpression.Method.DeclaringType != typeof(NpgsqlFullTextSearchDbFunctionsExtensions)
-            && methodCallExpression.Method.DeclaringType != typeof(NpgsqlFullTextSearchLinqExtensions)
-            && base.IsEvaluatableMethodCall(methodCallExpression);
+        /// <summary>
+        /// The static method info for <see cref="IPAddress.Parse(string)"/>.
+        /// </summary>
+        [NotNull] static readonly MethodInfo IPAddressParse =
+            typeof(IPAddress).GetRuntimeMethod(nameof(IPAddress.Parse), new[] { typeof(string) });
+
+        /// <summary>
+        /// The static method info for <see cref="PhysicalAddress.Parse(string)"/>.
+        /// </summary>
+        [NotNull] static readonly MethodInfo PhysicalAddressParse =
+            typeof(PhysicalAddress).GetRuntimeMethod(nameof(PhysicalAddress.Parse), new[] { typeof(string) });
+
+        /// <summary>
+        /// The static method info for <see cref="NpgsqlTsQuery.Parse(string)"/>.
+        /// </summary>
+        [NotNull] static readonly MethodInfo TsQueryParse =
+            typeof(NpgsqlTsQuery).GetRuntimeMethod(nameof(NpgsqlTsQuery.Parse), new[] { typeof(string) });
+
+        /// <summary>
+        /// The static method info for <see cref="NpgsqlTsVector.Parse(string)"/>.
+        /// </summary>
+        [NotNull] static readonly MethodInfo TsVectorParse =
+            typeof(NpgsqlTsVector).GetRuntimeMethod(nameof(NpgsqlTsVector.Parse), new[] { typeof(string) });
+
+        /// <inheritdoc />
+        public NpgsqlEvaluatableExpressionFilter([NotNull] IModel model) : base(model) {}
+
+        /// <inheritdoc />
+        public override bool IsEvaluatableMethodCall(MethodCallExpression methodCallExpression)
```

Comment:
As explained in https://github.com/npgsql/Npgsql.EntityFrameworkCore.PostgreSQL/issues/534#issuecomment-406946374, we shouldn't be touching `NpgsqlEvaluatableExpressionFilter` at all.